### DISCUSSION TITLE
Gridsome and Pinia results in undefined reading '_s'

### DISCUSSION THREAD

**Discussion by cherimo** [vuejs/pinia#2015] [Help and Questions]
### Reproduction

Not possible at the moment

### Steps to reproduce the bug

I have a working Gridsome application with Pinia Store.
I have implemented in the following way.

Default.Vue
```
<div>
  <input type="button" value="test" @click="updatePin" />
  <div id="store">store: {{ store.someState }} --</div>
</div>
</template>

<script>
import {useStore} from "../../state"

export default {
  name: "LocaleSwitcher",
  data: function () {
    return {
      store: useStore(),
    }
  },
  methods: {
    updatePin () {
      this.store.someState = "check"
    }
  }
}
</script>
```

state.js
```
import { defineStore } from 'pinia'

export const useStore = defineStore('store', {
  state: () => {
    return {
      someState: 'hello pinia',
    }
  },
  persist: true,
})
```

### Expected behavior

That Pinia works the same as in development mode where I can save and retrieve data.

### Actual behavior

The Pinia store works fine in Development mode but as soon as I tried to build the project I get the following error:
```
TypeError: Cannot read properties of undefined (reading '_s')
    at useStore (node_modules/pinia/dist/pinia.mjs:1697:0)
    at i.data (src/layouts/Default.vue:26:13)
```

**Comment by xiangnanscu** [comment]
I also Came across this issue

**Comment by vallemar** [comment]
I have same error. What is the problem?

**Comment by Mallfurion** [comment]
any idea what is causing this?
I am also experiencing this in production upon page reload.

**Reply by vallemar** [reply]
the problem is the monorepos, I have fix temporary this with https://github.com/vuejs/pinia/discussions/1073#discussioncomment-5272849

main.js

```ts
const pinia = createPinia()
app.use(pinia)
window.pinia = pinia
```
component
```ts
<script setup>
const xyzStore = useXYZStore(window.pinia)
</script>
```
